uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/api/v3/LeaveRequestTest.php

```diff
@@ -2038,4 +2040,310 @@ private function createLeaveRequestDatesArray(LeaveRequest $leaveRequest) {
 
     return $dates;
   }
+
+  /**
+   * @expectedException CiviCRM_API3_Exception
+   * @expectedExceptionMessage Mandatory key(s) missing from params array: leave_request_id
+   */
+  public function testAddCommentShouldThrowAnExceptionIfLeaveRequestIDIsMissing() {
+    civicrm_api3('LeaveRequest', 'addcomment', [
+      'text' => 'Random Commenter',
+      'contact_id' => 1
+    ]);
+  }
+
+  /**
+   * @expectedException CiviCRM_API3_Exception
+   * @expectedExceptionMessage Mandatory key(s) missing from params array: contact_id
+   */
+  public function testAddCommentShouldThrowAnExceptionIfContactIDIsMissing() {
+    civicrm_api3('LeaveRequest', 'addcomment', [
+      'leave_request_id' => 1,
+      'text' => 'Random Commenter',
+    ]);
+  }
+
+  /**
+   * @expectedException CiviCRM_API3_Exception
+   * @expectedExceptionMessage Mandatory key(s) missing from params array: text
+   */
+  public function testAddCommentShouldThrowAnExceptionIfTextIsMissing() {
+    civicrm_api3('LeaveRequest', 'addcomment', [
+      'leave_request_id' => 1,
+      'contact_id' => 1,
+    ]);
+  }
+
+  public function testAddCommentCanCreateAndUpdateCommentForLeaveRequest() {
+    $result = civicrm_api3('LeaveRequest', 'addcomment', [
```

Comment:
Add comment should not be able to update comments. In fact, updating comments is not something that is possible in L&A (at least for now)